internal/controller/serverset/nic_controller.go

```diff
@@ -149,9 +151,17 @@ func fromServerSetToNic(cr *v1alpha1.ServerSet, name, serverID, lanID string, re
 				LanCfg: v1alpha1.LanConfig{
 					LanID: lanID,
 				},
+				Dhcp: serverSetNic.DHCP,
```

Comment:
can dhcp change? If so, we need to add the update logic or a todo. If not, we can add a validation that it's immutable.

Reply:
In Cloud API it can be updated, but I am not sure if we want to allow it or not

Reply:
Why should we be bothered with this? I remember that we said that StatefulServerSet NICs are immutable: https://github.com/ionos-cloud/crossplane-provider-ionoscloud/blob/feat/serverset_ctrl/apis/compute/v1alpha1/serverset_types.go#L78

Reply:
forgot, great!